src/network/networkctl.c

```diff
@@ -636,6 +636,10 @@ static int next_lldp_neighbor(FILE *f, sd_lldp_neighbor **ret) {
         if (l != sizeof(u))
                 return -EBADMSG;
 
+        /* each LLDP TLV is 0-511 octets in size. */
+        if (le64toh(u) < 0 || le64toh(u) >= 512)
+                return -EBADMSG;
```

Comment:
hmm, each file contains one LLDP PDU, not just one LLDP TLV. Or in non-lldp speak: one full LLDP datagram. On ethernet that's something like 1522 bytes max. That said, the source here is trusted, and the LLDP protocol doesn't really care about the length much. Hence I think I'd rather check for a size limit of 4K or so (with a comment that 4K is substantially larger than the max Ethernet MTU, which is the size that matters).

Oh, and le64_t is unsigned, hence the left hand size check is unnecessary... (and the compiler should warn about that even, no?)

Reply:
> Oh, and le64_t is unsigned, hence the left hand size check is unnecessary... (and the compiler should warn about that even, no?)

Ah, indeed, sorry about that one, don't know how I missed the warning...

> hmm, each file contains one LLDP PDU, not just one LLDP TLV. Or in non-lldp speak: one full LLDP datagram. 

Oh I see. That was my first impression, but then I saw iteration and next and I started thinking it was a TLV...

Ok updated to 4096, but should we worry about MTU like jumbo frames where this would be 9000 (or 9022)? In theory we could even go with 64KiB which is what the 16-bit field in the Ethernet frame can cope with, right?

Reply:
so, i tried to figure that out earlier, but from all i gathered lldp is never sent as jumbo frames, so i think 4K should be ample...

Reply:
Pushed an update... Please take another look.